### ISSUE TITLE
Directly access serivce layer databases in resources now being synced, get rid of Sync job

### ISSUE THREAD

**Issue by LemonAndroid** [LemonAndroid/natural-backend#31]
The table, row, column and rowvalue resources need to directly access the associated database, and the intermediary database layer in the natural database needs to be dropped for the associated models.

Being dependent on the Sync job is adding unneeded complexity and causes issues, since for every possible syncing scenario there needs to be logic and tests for that logic.

The controllers and models handling the functionality for displaying tables, rows and row values must directly access the service layer database (the database that is created by the user).

**Comment by mul1sh** [comment]
@LemonAndroid sorry this may be a dump question, but just to ask, why was the sync job logic added in the first place?

**Comment by LemonAndroid** [comment]
In order to sync the tables, rows, columns (and the values) that are created visually through the [ember frontend](/LemonAndroid/natural-frontend) with the database that is accesible via the API or by the `Database#postgres_url`.

The reason it is implemted that way is purely pragmatically, since I started the app by scaffolding.

**Comment by mul1sh** [comment]
@LemonAndroid awesome thanks 👍  let me sort this then

**Comment by LemonAndroid** [comment]
@mul1sh In case there is even small progress, please create a PR so our ideas of the implementation are aligned.

**Comment by mul1sh** [comment]
@LemonAndroid  will do, about to do my first PR in a few hours 😃

**Comment by LemonAndroid** [comment]
@mul1sh Any progress?

**Comment by mul1sh** [comment]
@LemonAndroid yes though I got delayed a bit by the authentication /token route. Basically i was able to get a token but trying to access any of the other routes i.e. columns, i was getting the error
`{
    "error": "Not Authenticated"
}` . Anyway so for now i disabled authentication and i'm done with the /columns and /databases routes, but now i'm stuck again with the /projects route but i'm now stuck again with the `authenticate_project` method :(. So yeah thats why i'm basically at now

**Comment by LemonAndroid** [comment]
Please put up a PR with changes, that'll make things easier.

**Comment by mul1sh** [comment]
@LemonAndroid ok creating one now

**Comment by mul1sh** [comment]
@LemonAndroid  actually sorry I've seen the issue with the Authentication and got it working. I was using the name `Authorization` instead of `Authentication` 🤦‍♂️ , i guess i'm used to using the header as `Authorization`.

So anyway i've forked the repo and redoing my changes with Authentication enabled, I think it will be much easier than undoing my changes for the last 2 days. So anyway i'm still doing a PR in less than an hour, thanks

**Comment by LemonAndroid** [comment]
@mul1sh Why give timeframes when they are not accurate? just don't give them.

**Comment by mul1sh** [comment]
@LemonAndroid sorry for that 😞 , restoring my changes took some bit of time. Kindly let me alert you once i'm done

**Comment by LemonAndroid** [comment]
@mul1sh Sure, it's more than fine, take all the time needed.

**Comment by mul1sh** [comment]
@LemonAndroid awesome thanks and btw do you have like a skype or slack where we can chat? otheriwise this comment section will full up because I have a lot of questions coming up 😃

**Comment by LemonAndroid** [comment]
Sure, i'll create a gitter for this repo and we can also use the private messaging on gitter.

**Comment by mul1sh** [comment]
Cool thanks, kindly alert me when it ready

**Comment by LemonAndroid** [comment]
https://gitter.im/natural-backend/Lobby

**Comment by LemonAndroid** [comment]
I will refund this bounty by claiming it myself.
